### PR TITLE
OPENNLP-1592 Provide tests for MetaSummarizer

### PR DESCRIPTION
**Pull request by mawiesne**
This PR
-
- provides a minimal set of JUnit tests for MetaSummarizer
- improves the existing code so that summarization works (again)
- improves the code quality
- adds some JavaDoc where possible.

Tasks
-
Thank you for contributing to Apache OpenNLP.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with OPENNLP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn clean install` at the root opennlp-sandbox folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file in opennlp-sandbox folder?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found in opennlp-sandbox folder?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: summarizer/src/main/java/opennlp/summarization/Sentence.java [hunk 1]
```diff
@@ -135,20 +131,25 @@ public int getWordCnt()
...
 	@Override
-	public boolean equals(Object o) {
+	public final boolean equals(Object o) {
 		if (this == o) return true;
-		if (o == null || getClass() != o.getClass()) return false;
-		Sentence sentence = (Sentence) o;
-		return sentId == sentence.sentId;
+		if (!(o instanceof Sentence sentence)) return false;
+
+    return sentId == sentence.sentId;
```
Comment (kinow) [inline summarizer/src/main/java/opennlp/summarization/Sentence.java:140]: On GH UI it's showing the indentation a bit messed up. But maybe it's just GH's UI...

**Review by kinow — APPROVED** [review]

File: summarizer/src/main/java/opennlp/summarization/Sentence.java (hunk 1 above)
Comment (mawiesne) [reply summarizer/src/main/java/opennlp/summarization/Sentence.java:140]: Good pointer, will look at it and auto-format all classes in that package.

**Comment by rzo1** [comment]
Wondering if checkstyle is active for sandbox? Might be good to have it for future PRs.

**Review by rzo1 — APPROVED** [review]

**Comment by mawiesne** [comment]
> Wondering if checkstyle is active for sandbox? Might be good to have it for future PRs.

That's a separate issue for this. But worth introducing.

**Pull request merged**